sign1.go

```diff
@@ -110,6 +110,12 @@ func (m *Sign1Message) UnmarshalCBOR(data []byte) error {
 }
 
 // Sign signs a Sign1Message using the provided Signer.
+// The signature is stored in m.Signature.
+//
+// Note that m.Signature is only valid as long as m.Headers.Protected and
+// m.Payload remain unchanged after calling this method.
```

Comment:
another condition that _may be_ worth to flag is when `external` has changed

Reply:
`external` is not kept in any `Sign1Message` field but encoded as part of the signature, so modifying `external` after calling `Sign1` should be a problem.

Reply:
yes, I'm not sure here we want to highlight all possible ways the computed signature can be invalidated, or only those conditions that depend on the state of the message that we store.  That's why I left it to your judgement :-)